modules/gltf/gltf_document.cpp

```diff
@@ -4464,7 +4464,26 @@ Error GLTFDocument::_parse_texture_samplers(Ref<GLTFState> p_state) {
 		} else {
 			sampler->set_wrap_t(GLTFTextureSampler::WrapMode::DEFAULT);
 		}
-
+		if (d.has("extensions")) {
+			const Dictionary &extensions = d["extensions"];
+			if (extensions.has("GODOT_texture_sampler_anisotropic")) {
+				const Dictionary &anisotropic_ext = extensions["GODOT_texture_sampler_anisotropic"];
+				if (anisotropic_ext.has("anisotropic_filter_level")) {
+					const int filter_level = anisotropic_ext["anisotropic_filter_level"];
+					if (filter_level > 1) {
+						if (sampler->get_min_filter() == GLTFTextureSampler::FilterMode::NEAREST_MIPMAP_LINEAR || sampler->get_min_filter() == GLTFTextureSampler::FilterMode::NEAREST_MIPMAP_NEAREST) {
+							sampler->set_min_filter(GLTFTextureSampler::FilterMode::NEAREST_MIPMAP_ANISOTROPIC);
+						} else {
+							sampler->set_min_filter(GLTFTextureSampler::FilterMode::LINEAR_MIPMAP_ANISOTROPIC);
+						}
+					}
+				}
+			}
+		}
+		sampler->set_min_filter(p_state->get_texture_filter_min());
```

Comment:
This PR doesn't make any sense for multiple reasons:

* You are overriding the values set in the above `if` statements always, making the `if` statements useless.
* These are settings already specified each texture sampler, so there is no need for these settings in GLTFState.
* The code in GLTFState and GLTFTextureSampler is, for the most part, meant to represent data in the glTF file. If the intention is to override what is specified in the glTF file, that should be done in ResourceImporterScene.